### ISSUE TITLE
External authentication provider OAuth with grant_type=password

### ISSUE THREAD

**Issue by PavelPisotskiy** [OrchardCMS/OrchardCore#4969]
Hello, Could you please give an advice how to add external authentication provider (OAuth) with [grant_type=password](https://oauth.net/2/grant-types/password/). I need that users to be able to log in using an external user credentials (login and password) from the main login screen.

**Comment by sebastienros** [comment]
@MichaelPetrinolis do we have some doc to set it up?

@PavelPisotskiy have you tried to enable the existing modules we have , like Google, Twitter, ... they will show the auth button directly in the login form.

**Comment by MichaelPetrinolis** [comment]
Unfortunately we don't support this. Password flow is usually used by apps that are trusted by the Idp ( apps that are internal in the enterprise) as the user enters his username and password in these apps. This would require a custom account controller or modifications to the existing that would post the username/password to the Idp, get back the token and ignore local users (or match somehow the ext user with an internal one). It would be best to use one of the supported flows.

**Comment by PavelPisotskiy** [comment]
@sebastienros Yes, I've tried to enable the existing modules, and I've investigated how they work. But I need to integrate OC with internal system. This internal system provides OAuth authentication with grant_type=password. And I would like to log in there using standard form.
